File: Task/Web-scraping/Ruby/web-scraping-1.rb
require "open-uri"

open('http://tycho.usno.navy.mil/cgi-bin/timer.pl') do |p|
  p.each_line do |line|
    if line =~ /UTC/
      puts line.match(/ (\d{1,2}:\d{1,2}:\d{1,2}) /)
      break
    end
  end
end

File: Task/Web-scraping/Java/web-scraping-1.java
String scrapeUTC() throws URISyntaxException, IOException {
    String address = "http://tycho.usno.navy.mil/cgi-bin/timer.pl";
    URL url = new URI(address).toURL();
    try (BufferedReader reader = new BufferedReader(new InputStreamReader(url.openStream()))) {
        Pattern pattern = Pattern.compile("^.+? UTC");
        Matcher matcher;
        String line;
        while ((line = reader.readLine()) != null) {
            matcher = pattern.matcher(line);
            if (matcher.find())
                return matcher.group().replaceAll("<.+?>", "");
        }
    }
    return null;
}
